scenarios.md

```diff
@@ -0,0 +1,190 @@
+  * [AWS Container Registry (ecr)][ecr] - Omar Paul <omarpaul@amazon.com>
+  * [Docker Hub][docker-hub]
+  * [Google Container Registry (gcr)][gcr]
+  * [GitHub Package Registry (gpr)][gpr]
+  * [Quay][quay]
+  * [IBM Cloud Container Registry (icr)][icr]
+* Registry Vendors, Projects & Products
+  * [Docker Trusted Registry][docker-dtr]
+  * [Harbor][harbor]
+  * [JFrog Artifactory][jfrog]
+* Artifact Types
+  * Container Images - OCI Org
+  * Helm Charts
+  * Singularity
+  * Operator Bundles
+
+## Scenarios
+
+### Scenario #1: Local Build, Sign, Validate
+
+Prior to doing any deployment, a developer can test the: build, sign, validate scenario.
+
+1. Locally build a container image using a non-registry specific `name:tag`, such as:  
+  `$docker build hello-world:1`
+1. Locally sign `hello-world:1` 
+2. Run the image in a local environment which was configured to only support signed artifacts. *(**note** - details on the usability is deferred from this requirement. This could be a local docker host, a local kind or minikube instance configured to only run signed images)*  
+  `$ docker run hello-world:1`
+
+**Implications of this requirement:**
+
+- The local developer has access to the signing keys
+- The local environment has a policy by which it states the set of signatures it accepts.
+- The signing and validation of artifacts does not require a registry.
+- The signature used for validation may be hosted in a registry, or other accessible location.
+
+### Scenario #2: Sign, Rename, Push, Validate
+
+Once the developer has locally validated the build, sign, validate scenario, they will likely want to push the artifact to a registry used for deployment.
+
+1. Locally build and sign an artifact, such as a container image
+1. Rename the artifact to reflect the registry it will be pushed to:  
+  `$ docker tag hello-world:1 myregistry.domain.com/marketing/web:1`
+  `$docker push myregistry.domain.com/marketing/web:1`
+1. Deploy the artifact to a cluster that requires signatures:  
+  `$kubectl apply -f kubedeploy.yaml`
+  > **Note:** the above example presumes `kubedeploy.yaml` references the myregistry.domain.com/marketing/web:1 image
+
+**Implications of this requirement:**
+
+- Signatures can be validated based on the tag referenced, however the signature is not tied to a specific name. The tag is a pointer to the signed content.
+- This does not preclude tag locking scenarios. If a tag is updated with another signed artifact, the tag is considered signed and follows the signature validation rules of the new signature. For example, if a base image `fx:1.0` is signed, and rebuilt with a patched version, the updated `fx:1.0` is new content, and a valid scenario.
+- If the tag for `fx:1.0` is redirected to another unsigned manifest, the signature validation of `fx:1.0` will fail as it's no longer signed. This is no different than signed binaries being updated or replaced on a users local computer.
```

Comment:
Hmm. I'm trying to understand how you're reading this. I wrote it with the assumption that a customer requires artifacts to be signed. In this case, one of two things could happen.
1. the new version of `fx:1.0` isn't signed, so it fails validation is all artifacts must be signed
1. the new version is signed with a different key, which isn't really covered here. 

I'll leave this open for now, and perhaps we can discuss more on the next call.

Reply:
My confusion might stem from "validation" here: which party is doing that validation?

My initial read was that a client is performing validation on the signature.  For the client to know that the new `fx:1.0` is invalid because it's unsigned, the client must have some way to know that a signature is expected.  That could be one of several situations:

* the client has been configured to expect all images to have signatures
* the client has been configured to expect `fx:1.0` specifically to have a signature
* the client retrieved a previous manifest named `fx:1.0` that was signed, and now expects future manifests by that name to have a signature

> the new version is signed with a different key, which isn't really covered here.

This might be worth covering in another scenario, though I would expect the answer to be client-dependent and related to a particular trust policy.

Reply:
I changed validated to verified.
You're teasing out some good questions. 
I imagined a scenario where the user can specify a list of signatures that are considered valid, or * for any signature. But, I didn't call out the scenario where the signature was required. I don't consider * to imply it must be signed, rather any signature would be valid. 
In the `fx:1.0` scenario, the customer could say they only accept signatures from the org that owns fx.
I was going to open an issue for the signature required scenario, but I think that's not something we need to make an open question. It seems like a completely valid scenario.

Reply:
> I imagined a scenario where the user can specify a list of signatures that are considered valid, or * for any signature. But, I didn't call out the scenario where the signature was required. I don't consider * to imply it must be signed, rather any signature would be valid.

We might be getting too far into the trust policy space here.  I think that we should be specifying the mechanisms for creating, storing, and distributing signatures as well as how the signatures can be validated, but not the specific logic of "should this signature be trusted given a particular policy".

It might be worth thinking about other ecosystems here.  For example, web browsers: different browsers implement different default policies (and many allow those policies to be further configured).  Some browsers may decide to trust a given CA, or decide to trust certificates generated by that CA only before a certain date, or decide only to trust certificates that have been published to a transparency log.  But these policies are not part of a spec; they're implementation decisions made by the browser implementers.

Reply:
Finding the balance between being able to enable policies of various products and specifying a policy is the goal. I've been trying to avoid how k8s, docker, or cloud/vendor specific implementations might work. Rather, here's the basic scenarios we're trying to enable.

Reply:
I removed the details on stable tags, and made a simpler reference in the new 6.2 mydb scenario.